sdk/azcore/poller.go

```diff
@@ -7,442 +7,66 @@
 package azcore
 
 import (
-	"bytes"
-	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"net/http"
-	"strconv"
-	"strings"
-	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pollers"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pollers/loc"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pollers/op"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
 )
 
 // NewPoller creates a Poller based on the provided initial response.
 // pollerID - a unique identifier for an LRO, it's usually the client.Method string.
 // NOTE: this is only meant for internal use in generated code.
-func NewPoller(pollerID string, resp *http.Response, pl Pipeline, eu func(*http.Response) error) (*Poller, error) {
+func NewPoller(pollerID string, resp *http.Response, pl runtime.Pipeline, eu func(*http.Response) error) (*Poller, error) {
```

Comment:
Does the customer call NewPoller?  If so I don't like that we expose "runtime" package to them.

Reply:
I was thinking about this during the refactor (I meant to talk to Jeff about this yesterday but forgot).  We want to expose the core and ARM `Poller` types via their "root" packages `azcore` and `azcore/arm` respectively.  The usual convention is to place their constructors next to them, however it's true that we don't expect customer to call them.  However, if we were to move them to `runtime`, it gets a little weird as we now have two sets of constructors, so we'd have to rename the ARM ones to avoid name collisions.

@JeffreyRichter thoughts on this?

Reply:
Customers should not be newing up a Poller directly. Since a Poller requires an XxxClient (to poll against), either a customer will call an XxxClient method to create the new poller or we would have some package-level function that takes an xxxClient and returns the Poller.

Reply:
This is done.  I also applied the same pattern to the creation of ARM pollers, so we have symmetry between the two and keeps the uncommon stuff out of the happy path.